spec/controllers/two_factor_authentication/otp_verification_controller_spec.rb

```diff
@@ -127,6 +127,7 @@
       before do
         sign_in_before_2fa
         controller.user_session[:mfa_selections] = ['sms']
+        subject.user_session[:new_device] = true
```

Comment:
It'd be nice to have at least one test where we can verify an expected `new_device: false` in the analytics.

If it were me, I'd have avoided all these overrides to `subject.user_session[:new_device]` within existing specs, assert the expected default of `new_device: false`, and then have a new `context 'with new device'` that explicitly tests for `new_device: true` in a single location.

Reply:
It was unclear to me how the test created a user session. Without the override it comes through as `new_device: nil`

Reply:
Updated the MFA specs with those ideas in mind.